Jenkinsfile.public

```diff
@@ -13,24 +13,71 @@ pipeline {
         NPM_CONFIG_CACHE = '/tmp/npm/cache'
     }
     stages {
-        stage('Build') {
+        stage('BuildDRPC') {
```

Comment:
Just a tip, it should possible to loop over build flags and spin up things on separate nodes.

Reply:
Can you explain more? I'm not sure what you mean by "separate nodes"? It sounds like you mean I can run tests simultaneously on two separate VMs with completely separate build environments -- which would be great, but I don't know how to do this. :smile: If someone can assist me, that would be great.

Reply:
Ping @stefanbenten :P

Reply:
But Jenkins has capabilities to run jobs across multiple servers:
* https://jenkins.io/doc/pipeline/examples/#parallel-multiple-nodes
* https://www.cloudbees.com/blog/parallelism-and-distributed-builds-jenkins

Might require some learning on how to work with Jenkins, but Stefan should be able to help you set it up from Jenkins configuration side.

Reply:
@egonelbre That does not work with our descriptive pipeline sadly. We either have to run all tests in sequence or run the builds in sequence.
I discussed with @isaachess that it might be easier to add a second build job to the jenkins master that watches a copied pipeline called Jenkinsfile.drpc or something.

Reply:
After this PR is merged, we can make it a dependency, that both build succeed.

Reply:
Something like this should be possible... but I'm not certain about the syntax.

```
def run_platform(label, goflags = "")
{
    pipeline
    {
        agent{
            ...
        }
        ...
    }
}

parallel ( failFast: false,
    Linux: { run_platform("Windows") },
    LinuxDRPC: { run_platform("Windows", "-tags=drpc") },
    Windows: { run_platform("Windows") },
)
```

Reply:
Alternatively at least have one Jenkinsfile invoke the other but with GOFLAGS environment set.

Reply:
Can you explain why that is preferable to what we've got right now, with two jenkinsfiles? Especially because there is not need to run all of the checks twice, such as linting and npm checks, for example.

Reply:
Because previously we had two Jenkinsfiles, one for gerrit one for public and they ended up getting out-dated. For specific things that don't need those tests, we can disable them.

However, for some time, it'll be probably fine to manage both, with the expectation that at somepoint, someone will forget to update one.

Reply:
That's fair, though I hope this only exists for a week or two at most while we transition. It's also true that we don't need them to mirror each other, so there will many edits that should only be made to one.